checker/check-request-parameter-pattern-added-or-changed_test.go

```diff
@@ -0,0 +1,78 @@
+package checker_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tufin/oasdiff/checker"
+	"github.com/tufin/oasdiff/diff"
+)
+
+// CL: Changing pattern of request parameters
+func TestRequestParameterPatternChanged(t *testing.T) {
+	s1, err := open("../data/checker/request_parameter_pattern_added_or_changed_base.yaml")
+	require.NoError(t, err)
+	s2, err := open("../data/checker/request_parameter_pattern_added_or_changed_base.yaml")
+	require.NoError(t, err)
+
+	s2.Spec.Paths["/test"].Post.Parameters[0].Value.Schema.Value.Pattern = "^[\\w\\s]+$"
+	d, osm, err := diff.GetWithOperationsSourcesMap(getConfig(), s1, s2)
+	require.NoError(t, err)
+	errs := checker.CheckBackwardCompatibilityUntilLevel(singleCheckConfig(checker.RequestParameterPatternAddedOrChangedCheck), d, osm, checker.WARN)
+	require.Len(t, errs, 1)
+	require.Equal(t, checker.ApiChange{
+		Id:          "request-parameter-pattern-changed",
+		Text:        "changed the pattern of the 'query' request parameter 'category' from '^\\w+$' to '^[\\w\\s]+$'",
+		Comment:     "This is a warning because it is difficult to automatically analyze if the new pattern is a superset of the previous pattern (e.g. changed from '[0-9]+' to '[0-9]*')",
+		Level:       checker.WARN,
+		Operation:   "POST",
+		Path:        "/test",
+		Source:      "../data/checker/request_parameter_pattern_added_or_changed_base.yaml",
+		OperationId: "",
+	}, errs[0])
+}
+
+// CL: Adding pattern to request parameters
+func TestRequestParameterPatternAdded(t *testing.T) {
+	s1, err := open("../data/checker/request_parameter_pattern_added_or_changed_revision.yaml")
+	require.NoError(t, err)
+	s2, err := open("../data/checker/request_parameter_pattern_added_or_changed_base.yaml")
+	require.NoError(t, err)
+
+	d, osm, err := diff.GetWithOperationsSourcesMap(getConfig(), s1, s2)
+	require.NoError(t, err)
+	errs := checker.CheckBackwardCompatibilityUntilLevel(singleCheckConfig(checker.RequestParameterPatternAddedOrChangedCheck), d, osm, checker.WARN)
+	require.Len(t, errs, 1)
+	require.Equal(t, checker.ApiChange{
+		Id:          "request-parameter-pattern-added",
+		Text:        "added the pattern '^\\w+$' to the 'query' request parameter 'category'",
+		Comment:     "This is a warning because it is difficult to automatically analyze if the new pattern is a superset of the previous pattern (e.g. changed from '[0-9]+' to '[0-9]*')",
```

Comment:
This message makes sense for the use-case of changing the regex.
I would recommend to adjust the explanation for the specific use-case of adding a regex.